File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripService.java
package org.craftedsw.tripservicekata.trip;

import java.util.ArrayList;
import java.util.List;
import org.craftedsw.tripservicekata.exception.UserNotLoggedInException;
import org.craftedsw.tripservicekata.user.User;

public class TripService {

  public List<Trip> getTripsByUser(User user, User loggedUser) throws UserNotLoggedInException {
    if (loggedUser == null) {
      throw new UserNotLoggedInException();
    }

    return user.isFriendsWith(loggedUser) 
        ? tripsBy(user) 
        : noTrips();
  }

  private ArrayList<Trip> noTrips() {
    return new ArrayList<Trip>();
  }

  protected List<Trip> tripsBy(User user) {
    return TripDAO.findTripsByUser(user);
  }

}

File: kotlin/trip-service-kata/src/main/kotlin/org/craftedsw/tripservicekata/trip/TripService.kt
package org.craftedsw.tripservicekata.trip

import org.craftedsw.tripservicekata.exception.UserNotLoggedInException
import org.craftedsw.tripservicekata.user.User
import org.craftedsw.tripservicekata.user.UserSession
import java.util.*

class TripService {

    fun getTripsByUser(user: User): List<Trip> {
        var tripList: List<Trip> = ArrayList<Trip>()
        val loggedUser: User? = UserSession.instance.loggedUser
        var isFriend: Boolean = false
        if (loggedUser != null) {
            for (friend in user.friends) {
                if (friend == loggedUser) {
                    isFriend = true
                    break
                }
            }
            if (isFriend) {
                tripList = TripDAO.findTripsByUser(user)
            }
            return tripList
        } else {
            throw UserNotLoggedInException()
        }
    }

}
